### PR TITLE
#344 Added parameter --all to list all dependencies including the ones that do not need upgrades

### PR DESCRIPTION
**Pull request by gigi81**
Added support for a command line parameter --all to list all dependencies including the ones that do not need upgrades as detailed in #344

Example

```bash
dotnet-outdated.exe -t --all
» aspnettest
  [net7.0]
  Microsoft.AspNetCore.OpenApi                    7.0.8 == 7.0.8
  Microsoft.Extensions.ApiDescription.Server [T]  6.0.5 -> 7.0.8
  Microsoft.OpenApi [T]                           1.4.3 -> 1.6.5
  Swashbuckle.AspNetCore                          6.4.0 -> 6.5.0
  Swashbuckle.AspNetCore.Swagger [T]              6.4.0 -> 6.5.0
  Swashbuckle.AspNetCore.SwaggerGen [T]           6.4.0 -> 6.5.0
  Swashbuckle.AspNetCore.SwaggerUI [T]            6.4.0 -> 6.5.0

Version color legend:
<red>   : Major version update or pre-release version. Possible breaking changes.
<yellow>: Minor version update. Backwards-compatible features added.
<green> : Patch version update. Backwards-compatible bug fixes.

You can upgrade packages to the latest version by passing the -u or -u:prompt option.
Elapsed: 00:00:01.7981915
```

### CODE REVIEW

File: src/DotNetOutdated/Program.cs [hunk 1]
```diff
@@ -108,6 +108,10 @@ internal class Program : CommandBase
         [Option(CommandOptionType.NoValue, Description = "Treat package source failures as warnings.", ShortName = "ifs", LongName = "ignore-failed-sources")]
         public bool IgnoreFailedSources { get; set; } = false;
 
+        [Option(CommandOptionType.NoValue, Description = "Include all dependencies in the report even the ones not outdated",
+            ShortName = "a", LongName = "all")]
+        public bool All { get; set; } = false;
```
Comment (slang25) [inline src/DotNetOutdated/Program.cs:113]:
```suggestion
        [Option(CommandOptionType.NoValue, Description = "Include all dependencies in the report even the ones not outdated",
            ShortName = "utd", LongName = "inlcude-up-to-date")]
        public bool IncludeUpToDate { get; set; } = false;

```

This PR looks great, my only suggestion is to change the name to be a little more obvious

Comment (gigi81) [reply src/DotNetOutdated/Program.cs]: @slang25 updated the PR as per your review. Also updated the README to include the new argument

**Review by slang25 — APPROVED** [review]
Brilliant, thanks @gigi81 for this great addition 😄

**Comment by prettybits** [comment]
@slang25 The release notes for this change for v4.5.3 still mention a new `--all` parameter to have been added as they are generated from the commit messages which hasn't been updated in this PR to reflect the new argument name. Could the GitHub Release notes be edited to mention the correct `-utd|--include-up-to-date` instead? I was slightly confused at first wanting to try this out before having a look a the `--help` output. ;)

**Comment by slang25** [comment]
Thanks @prettybits, I hadn't considered that. I've updated the release notes and the issue here.

**Pull request merged**